chore(docs): update Docusaurus config and Netlify settings

critesjosh opened the pull request:
- Adjusted Docusaurus config to include current version based on environment variable.
- Added ignoreFiles option to exclude specific protocol specs from versioned docs.
- Updated Netlify configuration to define build commands for production and deploy-preview contexts.
- Modified package.json scripts for improved build and serve commands, including environment-specific options.

alejoamiras commented on docs/package.json:
```diff
@@ -3,11 +3,10 @@
   "private": true,
   "scripts": {
-    "dev": "HOST=0.0.0.0 ENV=dev yarn preprocess && ENV=dev docusaurus start --host ${HOST:-localhost}",
-    "dev:netlify": "yarn netlify dev",
```
I wouldn't remove this as it helps test the brevo's email subscription locally :)

critesjosh committed:
chore(docs): update Docusaurus config and Netlify settings

- Adjusted Docusaurus config to include current version based on environment variable.
- Added ignoreFiles option to exclude specific protocol specs from versioned docs.
- Updated Netlify configuration to define build commands for production and deploy-preview contexts.
- Modified package.json scripts for improved build and serve commands, including environment-specific options.

Co-authored-by: Josh Crites <jc@joshcrites.com>
